client/validate.go

```diff
@@ -0,0 +1,126 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"golang.org/x/xerrors"
+)
+
+// ValidateApplyOptions carries options for ApplyValidationSet.
+type ValidateApplyOptions struct {
+	Mode     string
+	Sequence int
+}
+
+// ValidationSetResult holds information about a single validation set.
+type ValidationSetResult struct {
+	ValidationSet string `json:"validation-set,omitempty"`
+	Mode          string `json:"mode"`
+	Sequence      int    `json:"sequence,omitempty"`
+	Valid         bool   `json:"valid"`
+	// TODO: flags/states for notes column
+}
+
+type postData struct {
+	Action   string `json:"action"`
+	Mode     string `json:"mode,omitempty"`
+	Sequence int    `json:"sequence,omitempty"`
+}
+
+// ForgetValidationSet forgets the given validation set identified by account,
+// name and optional sequence (if non-zero).
+func (client *Client) ForgetValidationSet(account, name string, sequence int) error {
+	if account == "" || name == "" {
+		return xerrors.Errorf("cannot forget validation set without account and name")
+	}
+
+	data := &postData{
+		Action:   "forget",
+		Sequence: sequence,
+	}
+
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(data); err != nil {
+		return err
+	}
+	path := fmt.Sprintf("/v2/validation-sets/%s/%s", account, name)
+	if _, err := client.doSync("POST", path, nil, nil, &body, nil); err != nil {
+		return xerrors.Errorf("cannot forget validation set: %v", err)
```

Comment:
(nitpick/curiosity) Given that we use xerrors here, would it make sense to use `%w` instead of `%v` (here and below) to benefit from the wrapping of xerrors? We do this in other places in client but not super consistently it seems.

Reply:
Yes, I guess it makes sense, thanks.